### PR TITLE
[NUI] Make BindableObject.IsBound public.

### PR DESCRIPTION
**Pull request by huayongxu**
### API Changes ###

 - ACR:

### CODE REVIEW

**Review by elishateng — APPROVED** [review]
LGTM

**Review by xiaowei-guan — APPROVED** [review]
LGTM

**Review by AchoWang — APPROVED** [review]
Looks good to me.

File: src/Tizen.NUI/src/internal/Common/FriendAssembly.cs [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
 [assembly: InternalsVisibleTo("Tizen.TV.NUI, " + Tizen.NUI.PublicKey.TizenTV)]
 [assembly: InternalsVisibleTo("Tizen.TV.NUI.Component, " + Tizen.NUI.PublicKey.TizenTV)]
+[assembly: InternalsVisibleTo("Tizen.TV.FLUX.Component.Extension, " + Tizen.NUI.PublicKey.TizenTV)]
```
Comment (hinohie) [inline src/Tizen.NUI/src/internal/Common/FriendAssembly.cs:33]: Not an major. Wasnt it looks more beautiful if we define this line under the `Tizen.TV.FLUX.Component`, not the `Tizen.TV.NUI.Component`?

**Review by hinohie — APPROVED** [review]

**Review by wanchao-xu — APPROVED** [review]
Looks good to me

**Review by wb0716 — APPROVED** [review]
Good for me

**Review by hyue7 — APPROVED** [review]
looks good

**Review by xiaowei-guan — APPROVED** [review]
LGTM

**Comment by huiyueun** [comment]
Could you check hino's comment and this PR? and you need to merge PR.

File: src/Tizen.NUI/src/internal/Common/FriendAssembly.cs (hunk 1 above)
Comment (huayongxu) [reply src/Tizen.NUI/src/internal/Common/FriendAssembly.cs]: Done.

**Comment by huayongxu** [comment]
> Could you check hino's comment and this PR? and you need to merge PR.

Done.

**Review by huiyueun — APPROVED** [review]

**Comment by dongsug-song** [comment]
@huayongxu 
Hello, 
could you explain why Tizen.TV.FLUX.Component.Extension is required to set as friend assembly?
as far as I know, there has been no this kind of request from VD.
thank you.

**Comment by elishateng** [comment]
> @huayongxu Hello, could you explain why Tizen.TV.FLUX.Component.Extension is required to set as friend assembly? as far as I know, there has been no this kind of request from VD. thank you.

Hi Mr. SSong, this is request from VD FLUX team in SRCN.

**Comment by elishateng** [comment]
> @huayongxu Hello, could you explain why Tizen.TV.FLUX.Component.Extension is required to set as friend assembly? as far as I know, there has been no this kind of request from VD. thank you.

Hi Mr. SSong,

VD FLUX team want to use IsBinded in Tizen.TV.FLUX.Component.Extension, like 'itemView.IsBinded = true', then their following issue can be solved.

This is their issue:
1. itemView 's BindingContext set to  data-0 (itemsSource[0], and Checked in this data is true )      =>  has no issue, item's Checked set to true by binding
2. itemView 's BindingContext = null;
3. Set itemView.Checked = false; //  by C# code;
4.  itemView 's BindingContext set to  data-8（Checked in this data is true too) ;   => has Issue,  expected that the itemView's Checked can be set to true, but not.

Here is their reference codes:

```
<DataTemplate x:Key="ImageTileOverlayTemplate02">
    <flc:ImageTileOverlay 
        UnitSizeWidth="35"
        MainImageResourceURL="{Binding Path=ResourceURL}"
        MainText="{Binding Path=TextString}"
        Checked="{Binding Path=Checked}"        //  {Binding Path : Checked is in ItemData class}
        CheckBoxVisibility="True">
        <x:Arguments>
            <x:String>C_ImageTileOverlay_01</x:String>
        </x:Arguments>
        <flc:ImageTileOverlay.LayoutParam>
            <flc:LayoutItemParam WidthResizePolicy="Fixed" HeightResizePolicy="MatchParent"></flc:LayoutItemParam>
        </flc:ImageTileOverlay.LayoutParam>
    </flc:ImageTileOverlay>
</DataTemplate>

public class ItemData : INotifyPropertyChanged
{
    ...
    public bool Checked
    {
        get => isChecked;
        set
        {
            isChecked = value;
            NotifyPropertyChanged(nameof(Checked));
        }
    }

    private void NotifyPropertyChanged([CallerMemberName] string propertyName = "")
    {
        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
    }
    ...
}

itemView.BindingContext = itemsSource[itemIndex];      // itemView is DataTemplate instance,  and  itemsSource is ItemData list

```

**Comment by dongsug-song** [comment]
> > @huayongxu Hello, could you explain why Tizen.TV.FLUX.Component.Extension is required to set as friend assembly? as far as I know, there has been no this kind of request from VD. thank you.
> 
> Hi Mr. SSong,
> 
> VD FLUX team want to use IsBinded in Tizen.TV.FLUX.Component.Extension, like 'itemView.IsBinded = true', then their following issue can be solved.
> 
> This is their issue:
> 
> 1. itemView 's BindingContext set to  data-0 (itemsSource[0], and Checked in this data is true )      =>  has no issue, item's Checked set to true by binding
> 2. itemView 's BindingContext = null;
> 3. Set itemView.Checked = false; //  by C# code;
> 4. itemView 's BindingContext set to  data-8（Checked in this data is true too) ;   => has Issue,  expected that the itemView's Checked can be set to true, but not.
> 
> Here is their reference codes:
> 
> ```
> <DataTemplate x:Key="ImageTileOverlayTemplate02">
>     <flc:ImageTileOverlay 
>         UnitSizeWidth="35"
>         MainImageResourceURL="{Binding Path=ResourceURL}"
>         MainText="{Binding Path=TextString}"
>         Checked="{Binding Path=Checked}"        //  {Binding Path : Checked is in ItemData class}
>         CheckBoxVisibility="True">
>         <x:Arguments>
>             <x:String>C_ImageTileOverlay_01</x:String>
>         </x:Arguments>
>         <flc:ImageTileOverlay.LayoutParam>
>             <flc:LayoutItemParam WidthResizePolicy="Fixed" HeightResizePolicy="MatchParent"></flc:LayoutItemParam>
>         </flc:ImageTileOverlay.LayoutParam>
>     </flc:ImageTileOverlay>
> </DataTemplate>
> 
> public class ItemData : INotifyPropertyChanged
> {
>     ...
>     public bool Checked
>     {
>         get => isChecked;
>         set
>         {
>             isChecked = value;
>             NotifyPropertyChanged(nameof(Checked));
>         }
>     }
> 
>     private void NotifyPropertyChanged([CallerMemberName] string propertyName = "")
>     {
>         PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
>     }
>     ...
> }
> 
> itemView.BindingContext = itemsSource[itemIndex];      // itemView is DataTemplate instance,  and  itemsSource is ItemData list
> ```

Thank you very much for detailed explanation. I understand that this problem is caused from that `BindableObject.IsBinded` is `internal`. so could you make it as hidden API by adding `[EditorBrowsable(EditorBrowsableState.Never)]`? because I heard that abusing a friend assembly is not good. please check again. thank you.

**Comment by elishateng** [comment]
> > > @huayongxu Hello, could you explain why Tizen.TV.FLUX.Component.Extension is required to set as friend assembly? as far as I know, there has been no this kind of request from VD. thank you.
> > 
> > 
> > Hi Mr. SSong,
> > VD FLUX team want to use IsBinded in Tizen.TV.FLUX.Component.Extension, like 'itemView.IsBinded = true', then their following issue can be solved.
> > This is their issue:
> > 
> > 1. itemView 's BindingContext set to  data-0 (itemsSource[0], and Checked in this data is true )      =>  has no issue, item's Checked set to true by binding
> > 2. itemView 's BindingContext = null;
> > 3. Set itemView.Checked = false; //  by C# code;
> > 4. itemView 's BindingContext set to  data-8（Checked in this data is true too) ;   => has Issue,  expected that the itemView's Checked can be set to true, but not.
> > 
> > Here is their reference codes:
> > ```
> > <DataTemplate x:Key="ImageTileOverlayTemplate02">
> >     <flc:ImageTileOverlay 
> >         UnitSizeWidth="35"
> >         MainImageResourceURL="{Binding Path=ResourceURL}"
> >         MainText="{Binding Path=TextString}"
> >         Checked="{Binding Path=Checked}"        //  {Binding Path : Checked is in ItemData class}
> >         CheckBoxVisibility="True">
> >         <x:Arguments>
> >             <x:String>C_ImageTileOverlay_01</x:String>
> >         </x:Arguments>
> >         <flc:ImageTileOverlay.LayoutParam>
> >             <flc:LayoutItemParam WidthResizePolicy="Fixed" HeightResizePolicy="MatchParent"></flc:LayoutItemParam>
> >         </flc:ImageTileOverlay.LayoutParam>
> >     </flc:ImageTileOverlay>
> > </DataTemplate>
> > 
> > public class ItemData : INotifyPropertyChanged
> > {
> >     ...
> >     public bool Checked
> >     {
> >         get => isChecked;
> >         set
> >         {
> >             isChecked = value;
> >             NotifyPropertyChanged(nameof(Checked));
> >         }
> >     }
> > 
> >     private void NotifyPropertyChanged([CallerMemberName] string propertyName = "")
> >     {
> >         PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
> >     }
> >     ...
> > }
> > 
> > itemView.BindingContext = itemsSource[itemIndex];      // itemView is DataTemplate instance,  and  itemsSource is ItemData list
> > ```
> 
> Thank you very much for detailed explanation. I understand that this problem is caused from that `BindableObject.IsBinded` is `internal`. so could you make it as hidden API by adding `[EditorBrowsable(EditorBrowsableState.Never)]`? because I heard that abusing a friend assembly is not good. please check again. thank you.

Hi Mr. SSong,
you mean make  `BindableObject.IsBinded` as public api and add `[EditorBrowsable(EditorBrowsableState.Never)]` for it ?

**Comment by dongsug-song** [comment]
@elishateng 
Hello, yes right. if it is not solution, please let me know, I will approve this patch for this time.

**Comment by huayongxu** [comment]
> @elishateng Hello, yes right. if it is not solution, please let me know, I will approve this patch for this time.

This patch is updated to make BindableObject.IsBinded public.

**Pull request merged**